service/matching/tasklist/matcher.go

```diff
@@ -436,7 +436,7 @@ func (tm *TaskMatcher) UpdateRatelimit(rps *float64) {
 
 // Rate returns the current rate at which tasks are dispatched
 func (tm *TaskMatcher) Rate() float64 {
-	return tm.limiter.Limit()
+	return float64(tm.limiter.Limit())
```

Comment:
though I'd like to add this....... it leads to a massive explosion in "well this is also the same value, so change that too..." chain of changes, with no good place to cut things off.

I think changing all these is very much worth it, `float64` is rather obviously a worse type.  But I'm not tackling that right now.